tensorboard/plugins/beholder/im_util.py

```diff
@@ -78,36 +78,20 @@ def pad_to_shape(array, shape, constant=245):
 
   return np.pad(array, padding, mode='constant', constant_values=constant)
 
-# New matplotlib colormaps by Nathaniel J. Smith, Stefan van der Walt,
-# and (in the case of viridis) Eric Firing.
-#
-# This file and the colormaps in it are released under the CC0 license /
-# public domain dedication. We would appreciate credit if you use or
-# redistribute these colormaps, but do not impose any legal restrictions.
-#
-# To the extent possible under law, the persons who associated CC0 with
-# mpl-colormaps have waived all copyright and related or neighboring rights
-# to mpl-colormaps.
-#
-# You should have received a copy of the CC0 legalcode along with this
-# work.  If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
-colormaps = np.load('{}/colormaps.npy'.format(resources_path()))
-magma_data, inferno_data, plasma_data, viridis_data = colormaps
-
 
 def apply_colormap(image, colormap='magma'):
   if colormap == 'grayscale':
     return image
 
   data_map = {
-      'magma': magma_data,
-      'inferno': inferno_data,
-      'plasma': plasma_data,
-      'viridis': viridis_data,
+      'magma': colormaps.MAGMA_DATA,
+      'inferno': colormaps.INFERNO_DATA,
+      'plasma': colormaps.PLASMA_DATA,
+      'viridis': colormaps.VIRIDIS_DATA,
   }
 
   colormap_data = data_map[colormap]
-  return (colormap_data[image]*255).astype(np.uint8)
+  return (np.array(colormap_data)[image] * 255).astype(np.uint8)
```

Comment:
Hmm, I wonder if maybe it's worth doing the np.array() conversion while creating the constant itself.  That would avoid having to do it every time apply_colormap is called (aka every single frame).

Unlike tuples we can't quite make a numpy array immutable, but we could set the writable flag to false so that it at least couldn't be inadvertently modified: https://stackoverflow.com/questions/5541324/immutable-numpy-array

Reply:
Great idea. PTAL.

Reply:
Looks great!  And yeah, makes sense that we could do the * 255 and uint8 cast early too, nice idea.